src/Integration/LocalServices/UnboundProjectFinder.cs

```diff
@@ -44,9 +45,16 @@ public UnboundProjectFinder(IServiceProvider serviceProvider, ILogger logger)
 
         internal UnboundProjectFinder(IServiceProvider serviceProvider, ILogger logger, IProjectBinderFactory projectBinderFactory)
         {
-            this.serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
             this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
             this.projectBinderFactory = projectBinderFactory ?? throw new ArgumentNullException(nameof(projectBinderFactory));
+
+            configProvider = serviceProvider.GetService<IConfigurationProviderService>();
+            projectSystem = serviceProvider.GetService<IProjectSystemHelper>();
```

Comment:
Moved the `GetService` calls to the constructor - reduces the number of thread-switches that are required in the rest of the code.